scripts/plugins/firewall-port

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+#
+# Copyright (c) Citrix Systems 2017. All rights reserved.
+
+set -e
+
+# Use this script to open/close port. 
+
+
+OP="$1"
+PORT="$2"
+CHAIN="xapi-INPUT"
+RULE="-p tcp -m conntrack --ctstate NEW -m tcp --dport $PORT -j ACCEPT"
+
+case "${OP}" in
+    open)
+        if ! iptables -C $CHAIN $RULE > /dev/null 
+        then
+            if ! iptables -S $CHAIN > /dev/null 
+            then
+                iptables -N $CHAIN
+                iptables -I INPUT -j $CHAIN
+            fi
+        iptables -I $CHAIN $RULE
+        service iptables save
+        fi
+        ;;
+    close)
+        iptables -D $CHAIN $RULE
```

Comment:
If the rule does not exist then this attempt to delete it will fail, causing the whole script to fail (non-zero exit code) and an error message will be written to stderr.

On the other hand, if the rule exists already then calling the `open` command will do nothing successfully.

This is inconsistent.
What behaviour do we want if someone calls `open 6640` twice, or calls `close 6640` twice?
I don't know, but I think it should be the same for `open` and `close` unless there is a really good reason.